src/FluentValidation.DependencyInjectionExtensions/ServiceCollectionExtensions.cs

```diff
@@ -92,14 +93,14 @@ public static IServiceCollection AddValidatorsFromAssemblyContaining<T>(this ISe
 		bool shouldRegister = filter?.Invoke(scanResult) ?? true;
 		if (shouldRegister) {
 			//Register as interface
-			services.Add(
+			services.TryAddEnumerable(

```

Comment:
Why did you use TryAddEnumerable and not TryAdd here?

Reply:
The desired behavior is to register every validator both
- as itself (serviceType == implementationType) to be able to resolve as itself
- as `IValidator<Target>` (serviceType = `IValidator<Target>`, implementationType = itself) to be able to inject `IEnumerable<IValidator<Target>>`

The first use case is covered via `TryAdd`, since it ensures that only one instance per serviceType is being registered. Multiple calls to AddValidatorsFromAssembly will therefore not lead to registering validators twice.
The second use case needs `TryAddEnumerable`, since that method matches the combination of serviceType and implementationType. As the name suggests, you can register multiple implementation types (e.g. `SomePartialModelValidator`, `SomeOtherPartialModelValidator`) as the same serviceType (e.g. `IValidator<Model>`).
`TryAdd(interfaceType, validatorType)` would lead to only the first validator for a validation target being registered and every other being ignored.

Reply:
I think we need to be careful about using `TryAddEnumerable` here, typically you only want 1 validator to be registered for a specific type, so the behaviour of only having the first validator being registered and ignoring the rest might be preferable. What happens if you register multiple validators for the same type and then try and resolve just 1, does it return the first or throw an exception? If the former, then this is fine but if it throws then we'll need to use TryAdd instead

Reply:
ok perfect, that should be fine then